### ISSUE TITLE
[Bug] Calls opening add phonenumber wrong customer

### ISSUE THREAD

**Issue by Scifire** [voicefoundryap/amazon-connect-for-zendesk#82]
We have an issue that in rare cases an incoming call is written to the ticket which the agent is currently working on.

**Description:**
Agent is available and working on a Ticket from customer A.
Call from Customer B comes in and is written in the Ticket which the Agent is currently working on (from Customer A). Additional the number from customer B is written to the Zendesk User from Customer A.

**To Reproduce**
Unfortunately, we´re unable to reproduce this since it happens only once or twice a week. But can narrow it down to incoming calls are always callbacks, and in most cases the customer B is not known by Zendesk before.

**Expected behavior**
A new Ticket should be opened for the new call from Customer B. 

**Environment (please complete the following information):**
 - App version 2.2.2 & 2.2.5RC

**Comment by karl-mentzer-vf** [comment]
@Scifire can you confirm what version you are currently using?

**Comment by karl-mentzer-vf** [comment]
Hi @Scifire - this was found in 2.2.2 however should have been resolved in 2.2.5 as a private release. We can provide a version with debug logging enabled if you are willing to run install as a private app and share time stamps when this occurs. Please advise.

**Comment by Scifire** [comment]
@karl-mentzer-vf as written this is not resolved in 2.2.5RC which we´re using as private app. 
Yes please share the version with debug logging enabled that we can evaluate it. If this will work we can of course share debug information here.
